crates/uv/tests/sync.rs

```diff
@@ -2106,3 +2107,80 @@ fn no_build() -> Result<()> {
 
     Ok(())
 }
+
+#[test]
+fn build_system_requires() -> Result<()> {
+    let context = TestContext::new("3.12");
+
+    let build = context.temp_dir.child("build");
+    build.child("pyproject.toml").write_str(
+        r#"
+        [project]
+        name = "build"
+        version = "0.1.0"
+        requires-python = ">=3.12"
+        dependencies = ["typing-extensions>=3.10"]
+
+        [build-system]
+        requires = ["setuptools>=42"]
+        build-backend = "setuptools.build_meta"
+        "#,
+    )?;
+
+    build
+        .child("src")
+        .child("build")
+        .child("__init__.py")
+        .write_str(indoc! { r#"
+            def hello() -> str:
+                return "Hello, world!"
+        "#})?;
+    build.child("README.md").touch()?;
+
+    let pyproject_toml = context.temp_dir.child("pyproject.toml");
+    pyproject_toml.write_str(
+        r#"
+        [project]
+        name = "project"
+        version = "0.1.0"
+        requires-python = ">=3.12"
+        dependencies = ["iniconfig>1"]
+
+        [build-system]
+        requires = ["setuptools>=42", "build"]
+        build-backend = "setuptools.build_meta"
+
+        [tool.uv.workspace]
+        members = ["build"]
+
+        [tool.uv.sources]
+        build = { workspace = true }
```

Comment:
Nit: Why a workspace member instead of a local path? This seems a bit more complicated and I think we'd want to duplicate the test if we stop requiring a source definition here. Maybe worth adding both now?

Reply:
Good idea, we can test both.